web/app/habit/create/components/step1.tsx

```diff
@@ -53,6 +57,42 @@ export default function CreateStep1({
   setMinTime,
   setStep,
 }: Step1Props) {
+  const [dateError, setDateError] = useState('');
+
+  useEffect(() => {
```

Comment:
nit pick: just thought maybe this can make it cleaner

```
useEffect(() => {
  const newError = startDate >= endDate ? 'Start date must be before end date' : '';
  setDateError(prevError => prevError !== newError ? newError : prevError);
}, [startDate, endDate]);
```

Reply:
Hmm i'm not sure this is cleaner 👀 , i feel like setState() with prev introduce another level of complexity, also this made condition checks from 1 to 2.

will keep it the old way